SPDY/NSURLRequest+SPDYURLRequest.m

```diff
@@ -154,4 +188,20 @@ - (void)setSPDYBodyFile:(NSString *)SPDYBodyFile
     }
 }
 
+- (void)setSPDYURLSession:(NSURLSession *)SPDYURLSession
+{
+    [self spdy_setIndirectObject:SPDYURLSession forKey:@"SPDYURLSession"];
+}
+
+- (void)spdy_setIndirectObject:(id)object forKey:(NSString *)key
+{
+    if (object == nil) {
+        [SPDYProtocol removePropertyForKey:key inRequest:self];
+    } else {
+        NSString *contextString = [[NSUUID UUID] UUIDString];
+        objc_setAssociatedObject(contextString, @selector(spdy_indirectObjectForKey:), object, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
```

Comment:
Yes.
Session owns the Task
Task owns the Request
Request owns the Session

On task complete:
Session will release the Task
Which (if it deallocs) will release the Request
Which (if it deallocs) will release the Session

So the key is the task, not the session.  I think there is a retain cycle with the SPDYProtocol though retaining the Task.  Commented on that property in SPDYProtoco+Project.h